terraform/restore-database/root.tf

```diff
@@ -0,0 +1,50 @@
+resource "aws_ssm_parameter" "database_url" {
+  name      = "/${local.environment}/${local.db_name}/database/url"
+  type      = "SecureString"
+  value     = aws_rds_cluster.db_restore_cluster.endpoint
+  overwrite = true
+}
+
+resource "random_string" "identifier" {
+  length  = 4
+  upper   = false
+  special = false
+}
+
+resource "aws_rds_cluster" "db_restore_cluster" {
+  restore_to_point_in_time {
+    source_cluster_identifier  = var.cluster_identifier
+    restore_type               = "copy-on-write"
+    use_latest_restorable_time = var.restore_time == "" ? true : null
+    restore_to_time            = var.restore_time == "" ? null : var.restore_time
+  }
+  vpc_security_group_ids              = [data.aws_security_group.db_security_group.id]
+  engine                              = "aurora-postgresql"
+  iam_database_authentication_enabled = true
+  db_subnet_group_name                = data.aws_db_subnet_group.subnet_group.name
+  database_name                       = local.db_name
+  final_snapshot_identifier           = "restored-${var.database}-final-snapshot-${random_string.identifier.result}-${local.environment}"
+  tags                                = local.common_tags
+}
+
+resource "aws_rds_cluster_instance" "atabase_instance" {
+  count                = 1
+  identifier_prefix    = "db-postgres-instance-${local.environment}"
+  cluster_identifier   = aws_rds_cluster.db_restore_cluster.id
+  engine               = "aurora-postgresql"
+  engine_version       = "11.9"
```

Comment:
AWS upgrades the DBs periodically. Will it cause any issues if this is older than the prod DB? Should we make it a parameter, or make it match the prod DB version by reading it as a data source?

Reply:
It's a good point and it makes sense as a variable I think. I've added in a variable and put a bit in the README about how to get it.